src/finufft_core.cpp

```diff
@@ -858,38 +848,35 @@ int FINUFFT_PLAN_T<TF>::setpts(BIGINT nj, TF *xj, TF *yj, TF *zj, BIGINT nk, TF
     nf = nf1 * nf2 * nf3; // fine grid total number of points
     if (nf * batchSize > MAX_NF) {
       fprintf(stderr,
-              "[%s t3] fwBatch would be bigger than MAX_NF, not attempting malloc!\n",
+              "[%s t3] fwBatch would be bigger than MAX_NF, not attempting memory "
+              "allocation!\n",
               __func__);
       return FINUFFT_ERR_MAXNALLOC;
     }
-    fftPlan->free(fwBatch);
-    fwBatch = fftPlan->alloc_complex(nf * batchSize); // maybe big workspace
+    fwBatch.resize(nf * batchSize); // maybe big workspace
```

Comment:
Yes. It has all the properties of a normal `vector`, but whenever its internal storage is (re-)allocated, it will be aligned at the requested boundary.